sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/UpdateFields.scala

```diff
@@ -17,19 +17,61 @@
 
 package org.apache.spark.sql.catalyst.optimizer
 
-import org.apache.spark.sql.catalyst.expressions.UpdateFields
+import java.util.Locale
+
+import scala.collection.mutable
+
+import org.apache.spark.sql.catalyst.expressions.{Expression, UpdateFields, WithField}
 import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.internal.SQLConf
 
 
 /**
- * Combines all adjacent [[UpdateFields]] expression into a single [[UpdateFields]] expression.
+ * Optimizes [[UpdateFields]] expression chains.
  */
-object CombineUpdateFields extends Rule[LogicalPlan] {
-  def apply(plan: LogicalPlan): LogicalPlan = plan transformAllExpressions {
+object OptimizeUpdateFields extends Rule[LogicalPlan] {
+  val optimizeUpdateFields: PartialFunction[Expression, Expression] = {
+    case UpdateFields(structExpr, fieldOps)
+      if fieldOps.forall(_.isInstanceOf[WithField]) &&
+        fieldOps.map(_.asInstanceOf[WithField].name.toLowerCase(Locale.ROOT)).distinct.length !=
```

Comment:
In case of `case-sensitive` mode, this seems to allow unnecessarily computation. Can we improve this `if` statement to handle both `case-sensitive` and `case-insensitive` together?

Reply:
The `if` condition should cover both case-sensitive and case-insensitive cases now. I compare names in lowercase in the condition.

Reply:
No, what I mean is that we don't need to execute line 39~69 at all.

Reply:
For example, for case-sensitive case, `fieldOps.map(_.asInstanceOf[WithField].name).distinct.length != fieldOps.length` should be used.

Reply:
I see. Updated. Thanks.